datafusion/core/src/execution/context.rs

```diff
@@ -239,20 +241,8 @@ impl SessionContext {
     /// This method is `async` because queries of type `CREATE EXTERNAL TABLE`
```

Comment:
I wonder if this docstring needs to be updated

Reply:
I also recommend adding a note like:

```suggestion
    /// This method is `async` because queries of type `CREATE EXTERNAL TABLE`
    ///
    /// Note: This api implements DDL such as `CREATE TABLE` and `CREATE VIEW` with in memory
    /// default implementations. Please [`SessionState::create_logical_plan()`] which does not
    /// mutate the state based on such statements.
```